docs/stable/2-account.md

```diff
@@ -213,12 +220,193 @@ All messages sent are encrypted with the public key of the destination and signe
+
+```go
+[]byte{
+    0xe7, 0x01, 0x04,
+    0x26, 0x1c, 0x55, 0x67, 0x5e, 0x55, 0xff, 0x25,
+    0xed, 0xb5, 0x0b, 0x34, 0x5c, 0xfb, 0x3a, 0x3f,
+    0x35, 0xf6, 0x07, 0x12, 0xd2, 0x51, 0xcb, 0xaa,
+    0xab, 0x97, 0xbd, 0x50, 0x05, 0x4c, 0x6e, 0xbc,
+    0x3c, 0xd4, 0xe2, 0x22, 0x00, 0xc6, 0x8d, 0xaf,
+    0x74, 0x93, 0xe1, 0xf8, 0xda, 0x6a, 0x19, 0x0a,
+    0x68, 0xa6, 0x71, 0xe2, 0xd3, 0x97, 0x78, 0x09,
+    0x61, 0x24, 0x24, 0xc7, 0xc3, 0x88, 0x8b, 0xc6,
+}
+```
+
+### De/Compression Process Flow
+
+The node MUST be passed a `multicodec` identified public key, of the above supported types, encoded with a valid `multibase` identifier.
+
+This specification RECOMMENDs that the node also accept an encoding type parameter to encode the output data. This provides for the case where the user requires the de/compressed key to be in a different encoding to the encoding of the given key. 
+
+#### Compression Example
+
+A hexadecimal encoded secp256k1 public chat key typically is represented as below:
+
+```text
+0x04261c55675e55ff25edb50b345cfb3a3f35f60712d251cbaaab97bd50054c6ebc3cd4e22200c68daf7493e1f8da6a190a68a671e2d3977809612424c7c3888bc6
+``` 
+
+To be properly interpreted by the node for compression the public key MUST be prepended with the `multicodec` `uvarint` code `0xea 0x01` and encoded with a valid `multibase` encoding, therefore giving the following:
+
+```text
+fea0104261c55675e55ff25edb50b345cfb3a3f35f60712d251cbaaab97bd50054c6ebc3cd4e22200c68daf7493e1f8da6a190a68a671e2d3977809612424c7c3888bc6
+``` 
+
+If adhering to the specification recommendation to provide the user with an output encoding parameter, the above string would be passed to the node with the following `multibase` encoding identifier.
+
+In this example the output encoding is defined as `base58 bitcoin`.
+
+```text
+z
+``` 
+
+The return value in this case would be
+
+```text
+zQ3shPyZJnxZK4Bwyx9QsaksNKDYTPmpwPvGSjMYVHoXHeEgB
+```
+
+Which after `multibase` decoding can be represented in bytes as below:
+
+```go
+[]byte{
+    0xe7, 0x01, 0x02,
+    0x26, 0x1c, 0x55, 0x67, 0x5e, 0x55, 0xff, 0x25,
+    0xed, 0xb5, 0x0b, 0x34, 0x5c, 0xfb, 0x3a, 0x3f,
+    0x35, 0xf6, 0x07, 0x12, 0xd2, 0x51, 0xcb, 0xaa,
+    0xab, 0x97, 0xbd, 0x50, 0x05, 0x4c, 0x6e, 0xbc,
+}
+```
+
+#### Decompression Example
+
+For the user the decompression process is exactly the same as compression with the exception that the user MUST provide a compressed public key for decompression. Else the decompression algorithm will fail.
+
+For further guidance on the implementation of public key de/compression consult the [`status-go` implementation and tests](https://github.com/status-im/status-go/blob/c9772325f2dca76b3504191c53313663ca2efbe5/api/utils_test.go).  
+
 ## Security Considerations
 
 -
 
 ## Changelog
 
+### Version 0.4
+
+Released // TODO
+
+- Added details of public key compression and decompression
```

Comment:
The way the spec is currently writing this is currently a breaking change

Reply:
I'm going rewrite the language to make public key de/compression implementation a SHOULD and make all MUSTs conditional on if de/compression functionality is being implemented.

Reply:
Done